base_layer/wallet/src/tasks/utxo_scanning.rs

```diff
@@ -0,0 +1,651 @@
+                    .await?
+                    .unwrap_or_else(|| 0.into());
+                let height = self
+                    .get_metadata(ScanningMetadataKey::RecoveryHeight)
+                    .await?
+                    .unwrap_or(0);
+                self.set_metadata(ScanningMetadataKey::RecoveryHeight, height).await?;
+
+                let _ = self
+                    .wallet
+                    .db
+                    .clear_client_value(RECOVERY_HEIGHT_KEY.to_string())
+                    .await?;
+                let _ = self
+                    .wallet
+                    .db
+                    .clear_client_value(RECOVERY_NUM_UTXOS_KEY.to_string())
+                    .await?;
+                let _ = self
+                    .wallet
+                    .db
+                    .clear_client_value(RECOVERY_TOTAL_AMOUNT_KEY.to_string())
+                    .await?;
+                self.publish_event(UtxoScannerEvent::Progress(final_utxo_pos, final_utxo_pos));
+                self.publish_event(UtxoScannerEvent::Completed(
+                    total_scanned,
+                    num_recovered,
+                    total_amount,
+                    elapsed,
+                ));
+            },
+            UtxoScannerMode::Scanning => {},
+        }
+
+        Ok(())
+    }
+
+    async fn connect_to_peer(&mut self, peer: NodeId) -> Result<PeerConnection, WalletError> {
+        self.publish_event(UtxoScannerEvent::ConnectingToBaseNode(peer.clone()));
+        match self.wallet.comms.connectivity().dial_peer(peer.clone()).await {
+            Ok(conn) => Ok(conn),
+            Err(e) => {
+                self.publish_event(UtxoScannerEvent::ConnectionFailedToBaseNode {
+                    peer,
+                    num_retries: self.num_retries,
+                    retry_limit: self.retry_limit,
+                    error: e.to_string(),
+                });
+
+                Err(e.into())
+            },
+        }
+    }
+
+    async fn attempt_sync(&mut self, peer: NodeId) -> Result<(u64, u64, Duration), WalletError> {
+        let mut connection = self.connect_to_peer(peer.clone()).await?;
+
+        let mut client = connection
+            .connect_rpc_using_builder(BaseNodeSyncRpcClient::builder().with_deadline(Duration::from_secs(60)))
+            .await
+            .map_err(to_wallet_scanning_error)?;
+
+        let latency = client
+            .get_last_request_latency()
+            .await
+            .map_err(to_wallet_scanning_error)?;
+        self.publish_event(UtxoScannerEvent::ConnectedToBaseNode(
+            peer.clone(),
+            latency.unwrap_or_default(),
+        ));
+
+        let timer = Instant::now();
+        let mut total_scanned = 0u64;
+        loop {
+            let start_index = self.get_start_utxo_mmr_pos().await?;
+            let tip_header = self.get_chain_tip_header(&mut client, &peer).await?;
```

Comment:
Don't rely on the tip, use the last hash and height to know whether the base node has reorged away from our last point
```suggestion
            let (start_index, start_header_hash, start_header_height) = self.get_last_sync_point().await?;
            let header = self.get_chain_header(&mut client, &peer, start_header_height).await?;
            if header.hash() != start_header_hash {
             // chain has reorg-ed, move last_start_point back and check from there
             ....
             
```

Reply:
Changed this to get the fork header, if any and work from there